source/pages/Home/Home.tsx

```diff
@@ -119,25 +138,59 @@ export const Home = () => {
     return formatBalanceDecimals(fiatPriceValue, true);
   }, [fiatPriceValue, isTestnet, moreThanMillion]);
 
-  return (
-    <div className={`scrollbar-styled h-full ${bgColor} overflow-auto`}>
-      {accounts[activeAccount.type][activeAccount.id] &&
+  const handleOnCloseFaucetModal = useCallback(() => {
+    wallet.setFaucetModalState(activeNetwork.chainId);
+  }, [activeNetwork, setFaucetModalState]);
+
+  const shouldShowFaucetFirstModal = useMemo(
+    () => faucetModal[activeNetwork.chainId],
+    [faucetModal, activeNetwork]
+  );
+
+  const formattedBalance = useMemo(
+    () =>
+      moreThanMillion
+        ? formatMillionNumber(actualBalance)
+        : formatBalanceDecimals(actualBalance || 0, false),
+    [actualBalance, moreThanMillion]
+  );
+
+  const shouldRenderHomePage = useMemo(
+    () =>
+      accounts[activeAccount.type][activeAccount.id] &&
       lastLogin &&
       isUnlocked &&
-      !isNetworkChanging ? (
+      !isNetworkChanging,
+    [accounts, activeAccount, lastLogin, isUnlocked, isNetworkChanging]
+  );
+
+  return (
+    <div className={`scrollbar-styled h-full ${bgColor} overflow-auto`}>
+      {shouldRenderHomePage ? (
         <>
           <Header accountHeader />
-          <WalletProviderDefaultModal />
+
+          {!isBitcoinBased &&
+            Object.values(FaucetChainIds).includes(activeNetwork.chainId) && (
```

Comment:
maybe this can be an external function

Reply:
fixed on https://github.com/syscoin/pali-wallet/pull/601